### PR TITLE
Supress unused var and state mut warnings for functions with empty body

### LINKED ISSUE #5295: Do not warn about functions with empty body.
Unused variable warning and state mutability warning should be suppressed for functions with empty body similar to how it is suppressed for functions without implementation.

### PR DESCRIPTION
**Pull request by leonardoalt**
Fixes #5295

### CODE REVIEW

File: libsolidity/analysis/StaticAnalyzer.cpp [hunk 1]
```diff
@@ -52,7 +52,10 @@ void StaticAnalyzer::endVisit(ContractDefinition const&)
 
 bool StaticAnalyzer::visit(FunctionDefinition const& _function)
 {
-	if (_function.isImplemented())
+	if (
+		_function.isImplemented() &&
+		!_function.body().statements().empty()
```
Comment (ekpyron) [inline libsolidity/analysis/StaticAnalyzer.cpp]: I'm not sure whether moving this check to the ``endVisit`` and just not issuing the warning in this case would be better.
But I think it's fine either way.

**Review by ekpyron — DISMISSED** [review]
Looks fine to me!

File: Changelog.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ### 0.5.2 (unreleased)
 
 Language Features:
+ * Do not warn about unused variables or state mutability for functions with an empty body.
```
Comment (chriseth) [inline Changelog.md]: Is "not warning about something anymore" really a language feature?

File: libsolidity/analysis/StaticAnalyzer.cpp (hunk 1 above)
Comment (chriseth) [reply libsolidity/analysis/StaticAnalyzer.cpp]: I think we should move it to endVisit. We might want to add other checks that do make sense if there is an empty body but not if there is no implementation.

**Review by chriseth — CHANGES_REQUESTED** [review]
Check should be moved to endVisit.

File: libsolidity/analysis/StaticAnalyzer.cpp (hunk 1 above)
Comment (leonardoalt) [reply libsolidity/analysis/StaticAnalyzer.cpp]: Moved to `endVisit`

File: Changelog.md (hunk 2 above)
Comment (leonardoalt) [reply Changelog.md]: Moved to `Compiler Features`

**Comment by leonardoalt** [comment]
Updated

**Review by chriseth — APPROVED** [review]

**Pull request merged**